openedx/core/djangoapps/credentials/management/commands/notify_credentials.py

```diff
@@ -164,23 +172,52 @@ def handle(self, *args, **options):
             'auto' if options['auto'] else 'manual',
         )
 
-        course_keys = self.get_course_keys(options['courses'])
-        if not (course_keys or options['start_date'] or options['end_date'] or options['user_ids']):
-            raise CommandError('You must specify a filter (e.g. --courses= or --start-date or --user_ids)')
+        program_course_run_keys = self._get_course_run_keys_for_programs(options["program_uuids"])
+
+        course_runs = options["courses"]
+        if not course_runs:
+            course_runs = []
+        if program_course_run_keys:
+            course_runs.extend(program_course_run_keys)
+
+        course_run_keys = self.get_course_keys(course_runs)
+        if not (
+            course_run_keys or
+            options['start_date'] or
+            options['end_date'] or
+            options['user_ids']
+        ):
+            raise CommandError(
+                'You must specify a filter (e.g. --courses, --program_uuids, --start-date, or --user_ids)'
+            )
 
-        handle_notify_credentials.delay(options, course_keys)
+        handle_notify_credentials.delay(options, course_run_keys)
 
-    def get_course_keys(self, courses=None):
+    def _get_course_run_keys_for_programs(self, uuids):
         """
-        Return a list of CourseKeys that we will emit signals to.
+        Retrieve all course runs for all of the given program UUIDs.
 
-        `courses` is an optional list of strings that can be parsed into
```

Comment:
Note: This comment was just wrong now. It looks like at one time it was true, but it was referencing how the management command used to work 2+ years ago.